File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.java
class Solution {
    public boolean halvesAreAlike(String s) {
		Set VOWELS = Set.of('a','e','i','o','u','A','E','I','O','U');
		int left = 0, right = s.length()-1;
		int lCount = 0, rCount = 0;
		while(left<right){
		    lCount += VOWELS.contains(s.charAt(left)) ? 1 : 0;
		    rCount += VOWELS.contains(s.charAt(right)) ? 1 : 0;
		    left++;
		    right--;
		}
		return lCount==rCount;        
    }
}

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        vowels = {"a","e","i","o","u"}
        v1 = list(filter(lambda x: x.lower() in vowels, s[:len(s)//2]))
        v2 = list(filter(lambda x: x.lower() in vowels, s[len(s)//2:]))
        return True if len(v1) == len(v2) else False